### PR TITLE
update benchmark configs with concurrent search scenarios

### PR DESCRIPTION
**Pull request by Gankris96**
### Description
Adding benchmarking config for concurrent segment search scenarios

### Related Issues
Helps to benchmark changes for https://github.com/opensearch-project/OpenSearch/issues/15259

### Check List
- [ ] ~Functionality includes testing.~
- [ ] ~API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.~
- [ ] ~Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Gankris96** [comment]
@jed326 @sohami Please help review.

File: .github/benchmark-configs.json [hunk 1]
```diff
@@ -167,5 +167,59 @@
       "data_instance_config": "4vCPU, 32G Mem, 16G Heap"
     },
     "baseline_cluster_config": "x64-r5.xlarge-single-node-1-shard-0-replica-baseline"
+  },
+  "id_11": {
+    "description": "Search only test-procedure for big5 with concurrent segment search setting enabled",
+    "supported_major_versions": ["2", "3"],
```
Comment (sohami) [inline .github/benchmark-configs.json:173]: i think this will only be 3 since snapshot used is created on 3.x version

Comment (Gankris96) [reply .github/benchmark-configs.json:173]: @sohami And do we want to set slice count to something or leave it default?

Comment (sohami) [reply .github/benchmark-configs.json]: we can leave it as is for now since we want to compare if there is any overhead of the default decider.

**Review by sohami — APPROVED** [review]

**Pull request merged**